glance/api/v2/images.py

```diff
@@ -548,15 +579,41 @@ def index(self, req, marker=None, limit=None, sort_key=None,
             self.policy.enforce(req.context, 'get_images', target)
 
             images = image_repo.list(marker=marker, limit=limit,
-                                     sort_key=sort_key,
-                                     sort_dir=sort_dir,
-                                     filters=filters,
-                                     member_status=member_status)
+                                    sort_key=sort_key,
+                                    sort_dir=sort_dir,
+                                    filters=filters,
+                                    member_status=member_status)
             db_image_count = len(images)
             images = [image for image in images
-                      if api_policy.ImageAPIPolicy(req.context, image,
-                                                   self.policy
-                                                   ).check('get_image')]
+                    if api_policy.ImageAPIPolicy(req.context, image,
+                                                self.policy
+                                                ).check('get_image')]
+
+            # CUSTOM: Determine effective domain name
+            domain_name = (
+                getattr(req.context, 'domain_name', None) or
+                getattr(req.context, 'user_domain_name', None)
+            )
+            LOG.debug("[IAAS_FILTER] Effective domain_name: %s", domain_name)
+
+            # CUSTOM: Enforce domain-scoped tag for iaas domains
+            if domain_name and domain_name.startswith('iaas'):
```

Comment:
Will probably need a filter for the special iaas domains. You might as well build it now. You could have used domain tags for enabling it on per-domain basis.